src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1477,6 +1507,12 @@ void DoCancelRequest(int backendProcessId, int backendSecretKey)
             }
         }
 
+        internal void ResetCancellation()
+        {
+            _cancellationRequested = false;
+            ReadBuffer.TimeoutCts.Reset();
```

Comment:
It seems like this is doing some unnecessary stuff (and is on the hot path, whenever a command is executed:
* Is _registration.Dispose needed? At that point when a command is executed, don't we know that there's no registration (because Stop was called before)?
* Same for CancelAfter(InfiniteTimeSpan) - wasn't it already done by the last Stop?
* The cancellationToken.CanBeCanceled check isn't needed, because we never pass in a token.

In other words, we have two uses of Reset: one when beginning to execute a new command, and one in the read buffer. We may want to split the two. Can also be done as a later optimization.

/cc @Brar

Reply:
Yeah, exactly - we shouldn't be doing anything else at this point.